integration-tests/src/tests/client/process_blocks.rs

```diff
@@ -5136,3 +5113,183 @@ mod new_contract_loading_cost {
+            nonce: 0,
+            block_hash: CryptoHash::default(),
+        };
+
+        let mut old_result = env.execute_tx(tx.clone(), &signer, epoch_length);
+        let old_outcome = &mut old_result.receipts_outcome[0].outcome;
+        // Overwrite receipt ids, those will change for many different reasons that are not relevant.
+        for receipt in &mut old_outcome.receipt_ids {
+            *receipt = CryptoHash::default();
+        }
+        insta::assert_debug_snapshot!(format!("{snap_name} before {feature:?}"), old_outcome);
+
+        env.upgrade_protocol(new_protocol_version);
+
+        let mut new_result = env.execute_tx(tx, &signer, epoch_length);
+        let new_outcome = &mut new_result.receipts_outcome[0].outcome;
+        for receipt in &mut new_outcome.receipt_ids {
+            *receipt = CryptoHash::default();
+        }
+        insta::assert_debug_snapshot!(format!("{snap_name} with {feature:?}"), new_outcome);
+    }
+
+    /// Test that current_gas_price host function returns the correct value.
+    #[test]
+    fn test_current_gas_price() {
+        let wat_code = &assert_gas_price_contract("current_gas_price", 10u64.pow(9));
+        let contract = wat::parse_str(wat_code).unwrap();
+        check_outcome_with_and_without_feature(
+            "current_gas_price",
+            Action::FunctionCall(FunctionCallAction {
+                method_name: "main".to_string(),
+                args: vec![],
+                gas: 3 * 10u64.pow(14),
+                deposit: 0,
+            }),
+            contract,
+            ProtocolFeature::GasPriceHostFn,
+        );
+    }
+
+    /// Test that purchased_gas_price host function returns the correct value.
+    #[test]
+    fn test_purchased_gas_price_300_tgas() {
+        // With 300 Tgas, pessimistic inflation is 1.03 ^ 61 = 6.06835119717
+        // and with some rounding it gets to 6_068_351_198.
+        let gas = 3 * 10u64.pow(14);
+        let expected_purchase_price = 6_068_351_198;
+        let wat_code = &assert_gas_price_contract("purchased_gas_price", expected_purchase_price);
+        let contract = wat::parse_str(wat_code).unwrap();
+        check_outcome_with_and_without_feature(
+            "300Tgas_purchased_gas_price",
+            Action::FunctionCall(FunctionCallAction {
+                method_name: "main".to_string(),
+                args: vec![],
+                gas,
+                deposit: 0,
+            }),
+            contract,
+            ProtocolFeature::GasPriceHostFn,
+        );
+    }
+
+    #[test]
+    fn test_purchased_gas_price_100_tgas() {
+        // With 100 Tgas, pessimistic inflation is 1.03 ^ 20 = 1.80611123467
+        // and with some rounding it gets to 1_806_111_235.
+        let gas = 10u64.pow(14);
+        let expected_purchase_price = 1_806_111_235;
+        let wat_code = &assert_gas_price_contract("purchased_gas_price", expected_purchase_price);
+        let contract = wat::parse_str(wat_code).unwrap();
```

Comment:
Let's use 

https://github.com/near/nearcore/blob/27d2e5acb67698c59a1bd5e7bbac1234f062195d/runtime/near-test-contracts/src/lib.rs#L9

here to abstract away which wasm parser we use